java/vector/src/main/java/org/apache/arrow/vector/ipc/ArrowWriter.java

```diff
@@ -103,6 +105,7 @@ public void start() throws IOException {
    */
   public void writeBatch() throws IOException {
     ensureStarted();
+    writeDictionaries();
```

Comment:
I would personally put the `if (!dictWritten)` check here, otherwise call `writeDictionaries` something like `ensureDictionariesWritten`

Reply:
Right, ensureDictionariesWritten looks better, thanks